### PR TITLE
User Guide to Hamilton Standard

### PR DESCRIPTION
**Pull request by xjjiang**
### Summary

This is the first draft of user guide of Hamilton Standard model.

### Related Issues

- Resolves #

### Backwards incompatibilities

None

### New Dependencies

None

### CODE REVIEW

File: aviary/docs/user_guide/hamilton_standard.md [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+Below is an XDSM diagram of Hamilton Standard model:
+
+![Hamilton Standard Diagram](images/hamilton_standard.png)
+
+The inputs are grouped in three aspects:
+
+Geometric inputs:
+- Propeller diameter
+- Activity factor per blade (range: 80 to 200, baseline: 150)
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:13]: Should "baseline" be "default"?

File: aviary/docs/user_guide/hamilton_standard.md [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+- Number of Blades (range: 2 to 8)
+
+Power inputs:
+- Shaft Horsepower to Propeller (hp)
+- Installation Loss Factor (0 to 1)
+
+Performance inputs:
+- Operating Altitude, ft
+- True Airspeed, knots
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:22]:
```suggestion
- Operating Altitude (ft)
- True Airspeed (knots)
```

File: aviary/docs/user_guide/hamilton_standard.md [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+
+The inputs are grouped in three aspects:
+
+Geometric inputs:
+- Propeller diameter
+- Activity factor per blade (range: 80 to 200, baseline: 150)
+- Number of Blades (range: 2 to 8)
+
+Power inputs:
+- Shaft Horsepower to Propeller (hp)
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:17]:
```suggestion
- Shaft power to Propeller (hp)
```

File: aviary/docs/user_guide/hamilton_standard.md [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+- Installation Loss Factor (0 to 1)
+
+Performance inputs:
+- Operating Altitude, ft
+- True Airspeed, knots
+- Propeller Tip Speed (Usually < 800 ft/s)
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:23]: is this maximum or instantaneous?
If it is max as set by a user, make a note of it.
If it is just current from rpm*radius, no change is needed.

File: aviary/docs/user_guide/hamilton_standard.md [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+Power inputs:
+- Shaft Horsepower to Propeller (hp)
+- Installation Loss Factor (0 to 1)
+
+Performance inputs:
+- Operating Altitude, ft
+- True Airspeed, knots
+- Propeller Tip Speed (Usually < 800 ft/s) 
+- Integrated Lift Coefficient (range: 0.3 to 0.8, baseline: 0.5)
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:24]: does "baseline" mean "default"

File: aviary/docs/user_guide/hamilton_standard.md [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+
+Note that some of the inputs are good for limited ranges. The Hamilton Standard model can have odd number of blades although the data provided are based on even number of blades. For odd number of blades, interpolations using 2, 4, 6 and 8 blade data are used. The corresponding outputs are:
+
+Geometric outputs:
+- Design Blade Pitch Angle (@0.75 Radius)
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:29]:
```suggestion
- Design Blade Pitch Angle (at 0.75 Radius)
```

File: aviary/docs/user_guide/hamilton_standard.md [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+
+`USatmos` component provides the flight condition. The flight condition is passed to `PreHamiltonStandard` component from which flight Mach number, propeller tip Mach number, advance ratio and power coefficient are computed. They are fed into `HamiltonStandard` component.
+
+![CP and CT matching](images/CPE_CTE_matching.png)
+
+HamiltonStandard is the core of the model. Given the power coefficient (CP) and advance ratio (J), it finds the blade angle (BL) by a CP-BL chart by tracing advance ratio and then with the blade angle, it finds the thrust coefficient (CT) using its CT-BL chart by tracing advance ratio again. This algorithm is shown in the above pair of charts. The CP → BL → CT chart matching algorithm is based on baseline data. If user inputs are not on baseline, it will first convert them to those baseline parameters by a sequence of interpolations to do corrections. The newly converted parameters are called “effective parameters” (e.g., CPE and CTE). The outputs are blade angle, thrust coefficient and tip compressibility loss factor.
+
+Finally, the thrust is computed in `PostHamiltonStandard` component based on thrust coefficient and tip compressibility loss factor.
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:60]:
```suggestion
Finally, the thrust is computed in the `PostHamiltonStandard` component based on thrust coefficient and tip compressibility loss factor.
```

File: aviary/docs/user_guide/hamilton_standard.md [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+
+Hamilton Standard model uses a bunch of wind tunnel test data from un-installed propellers. When a cell is mounted around the propeller, an installation loss factor is introduced. The installation loss factor can be given by user or computed. If it is computed, we need another group of components as shown below:
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:62]:
```suggestion
The Hamilton Standard model uses wind tunnel test data from un-installed propellers. When a nacelle is mounted behind the propeller, an installation loss factor is introduced. The installation loss factor can be given by the user or computed. If it is computed, we need another group of components as shown below:
```

File: aviary/docs/user_guide/hamilton_standard.md [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+Aircraft.Engine.PROPELLER_INTEGRATED_LIFT_COEFFICENT
+Aircraft.Engine.PROPELLER_ACTIVITY_FACTOR
+Aircraft.Engine.NUM_BLADES
+Aircraft.Design.COMPUTE_INSTALLATION_LOSS
+Dynamic.Mission.PROPELLER_TIP_SPEED
+Dynamic.Mission.SHAFT_POWER
+Dynamic.Mission.INSTALLATION_LOSS_FACTOR
+```
+
+To build a turboprop engine and an Aviary model, we use `TurboPropDeck` object with a special parameter `prop_model` set to `True`:
```
Comment (crecine) [inline aviary/docs/user_guide/hamilton_standard.md:81]:
```suggestion
To build a turboprop engine that uses the Hamilton Standard propeller model, we use a `TurboPropDeck` object with `prop_model` set to `True`:
```

**Review by crecine — COMMENTED** [review]
A couple of minor questions/suggestions, but overall this is looking good.

File: aviary/docs/user_guide/hamilton_standard.md (hunk 1 above)
Comment (xjjiang) [reply aviary/docs/user_guide/hamilton_standard.md:13]: Do you mean the word should be "default" or the default value should be 150? In Hamilton Standard, baseline means that the tables are provided based on that setting. I can change the actual default value to 150. Please let me know.

File: aviary/docs/user_guide/hamilton_standard.md (hunk 2 above)
Comment (xjjiang) [reply aviary/docs/user_guide/hamilton_standard.md:22]: Accept the suggestion.

File: aviary/docs/user_guide/hamilton_standard.md (hunk 5 above)
Comment (crecine) [reply aviary/docs/user_guide/hamilton_standard.md:24]: disregard, per response to other comment

File: aviary/docs/user_guide/hamilton_standard.md (hunk 3 above)
Comment (xjjiang) [reply aviary/docs/user_guide/hamilton_standard.md:17]: accepted.

File: aviary/docs/user_guide/hamilton_standard.md (hunk 4 above)
Comment (xjjiang) [reply aviary/docs/user_guide/hamilton_standard.md:23]: It is not the maximum tip speed.

File: aviary/docs/user_guide/hamilton_standard.md (hunk 6 above)
Comment (xjjiang) [reply aviary/docs/user_guide/hamilton_standard.md:29]: agree. done.

File: aviary/docs/user_guide/hamilton_standard.md (hunk 8 above)
Comment (xjjiang) [reply aviary/docs/user_guide/hamilton_standard.md:62]: Thanks for the modification. All are accepted.

File: aviary/docs/user_guide/hamilton_standard.md (hunk 9 above)
Comment (xjjiang) [reply aviary/docs/user_guide/hamilton_standard.md:81]: updated.

**Comment by crecine** [comment]
looks like the docs build is failing because hamilton_standard hasn't been added to the table of contents.

**Review by crecine — APPROVED** [review]

**Comment by xjjiang** [comment]
> looks like the docs build is failing because hamilton_standard hasn't been added to the table of contents.

Do I have to set a link to the table of contents? I am thinking that it is a sub item under another page (e.g. propulsion or turboprop).

**Comment by crecine** [comment]
> > looks like the docs build is failing because hamilton_standard hasn't been added to the table of contents.
> 
> Do I have to set a link to the table of contents? I am thinking that it is a sub item under another page (e.g. propulsion or turboprop).

I think it needs to appear somewhere in the table of contents for the doc builder to get the file.
Try adding `- file: user_guide/hamilton_standard` right after propulsion in the _toc.yml file.

```
- caption: User Guide
  chapters:
    - file: user_guide/user_interface
      sections:
      - file: user_guide/UI_Levels
      - file: user_guide/examples_of_the_same_mission_at_different_UI_levels
      - file: user_guide/aviary_commands
      - file: user_guide/input_files
    - file: user_guide/drawing_and_running_simple_missions
    - file: user_guide/pre_mission_and_mission
    - file: user_guide/outputs_and_how_to_read_them
    - file: user_guide/variable_metadata
    - file: user_guide/features_and_functionalities
      sections:
      - file: user_guide/aerodynamics
      - file: user_guide/external_aero
      - file: user_guide/propulsion
      - file: user_guide/hamilton_standard
```

**Comment by xjjiang** [comment]
Okay. I've added a link to the table of contents.

**Review by johnjasa — APPROVED** [review]

**Pull request merged**